docs/wiki/research/account-abstraction/eip-7702.md

```diff
@@ -7,15 +7,76 @@
+EIP-7702 proposes adding a new [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)-compatible transaction type that allows temporarily setting the code field for an EOA.  
+The new `TransactionType` is defined as `SET_CODE_TX_TYPE` (`0x04`).  
+The associated `TransactionPayload` is an RLP-serialized tuple containing the following fields:  
+`chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, value, data, access_list, authorization_list, signature_y_parity, signature_r, signature_s`, where the authorization must be a non-empty list of `chain_id, address, nonce, y_parity, r, s` tuples.   
+Here, the `address` refers to the address of the delegated contract code, and `chain_id` may be either a valid chain ID or 0, the `nonce` must match the EOA's nonce to provide replay protection and the `y_parity`, `r`, and `s` values are the ECDSA signature components signed by the `authority`.   
+Anyone can submit a transaction containing a valid authorization list — meaning the `tx.origin` does not have to be the EOA’s address, as long as the signature is valid. Setting the `chain_id` to `0` enables resubmission of the transaction across different chains, provided the nonce values match.  
+The `ReceiptPayload` is defined as the `status, cumulative_transaction_gas_used, logs_bloom, logs` tuple in RLP-seralized form.  
+More information about RLP encoding can be found in [this section](https://epf.wiki/#/wiki/EL/RLP).
+
+When such a transaction is submitted the following steps are performed for each tuple in the `authorization_list`:
+1. Basic validations are performed, such as verifying `chain_id < 2^256`, `len(auth.address) == 20` etc.. Additionally, the `chain_id` must either match the current chain's ID or be set to `0`.
+2. Verify that the `nonce` is less than `2^64 - 1`.
+3. The `authority` is recovered from the signature and the payload using the `ecrecover` method:  
+`authority = ecrecover(msg, y_parity, r, s).` where `msg = keccak(MAGIC || rlp([chain_id, address, nonce]))` and `s` must be less than `secp256k1n/2`.
+4. The recovered `authority` is marked as a warm address and is added to the `accessed_addresses` set.
+5. Verify that `nonce` is equal to  nonce of `authority` if the `authority` exists in the trie otherwise `nonce` has to equal `0`.
+6. If `authority` exists in the trie the refund amount of `PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST` is added to the global refund mechanism.
+7. In case of a non-zero `address` the code of the authority is set to `0xef0100 || address` (where `||` denotes concatenation), otherwise the EOA's code is cleared and the code hash is set to the hash of a null address (`0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`). This serves as a revocation mechanism.
+8. The nonce of `authority` is incremented by one.
```

Comment:
after this, it would be nice to know what happens if any of the steps above fail. from the EIP, "If any step above fails, immediately stop processing the tuple and continue to the next tuple in the list. When multiple tuples from the same authority are present, set the code using the address in the last valid occurrence.  Note, if transaction execution results in failure (e.g. any exceptional condition or code reverting), the processed delegation indicators is not rolled back"